### PR TITLE
Feature/bbb 186 refactor createuser

### PR DESCRIPTION
**Pull request by pauljroberts**

### CODE REVIEW

File: service/src/main/java/uk/gov/dft/bluebadge/service/usermanagement/service/UserManagementService.java [hunk 1]
```diff
@@ -49,14 +40,25 @@
    *
    * @param userEntity User to create.
    * @return Create count.
-   * @throws BlueBadgeBusinessException if validation fails.
+   * @throws BadResponseException if validation fails.
    */
-  public int createUser(UserEntity userEntity) throws BlueBadgeBusinessException {
+  public int createUser(UserEntity userEntity) {
     List<ErrorErrors> businessErrors = nonBeanValidation(userEntity);
```
Comment (robsvaltech) [inline service/src/main/java/uk/gov/dft/bluebadge/service/usermanagement/service/UserManagementService.java]: `List<ErrorErrors> businessErrors = businessValidate(userEntity);`

File: service/src/generated/java/uk/gov/dft/bluebadge/service/usermanagement/controller/UsersApi.java [hunk 2]
```diff
@@ -251,7 +293,7 @@
     produces = {"application/json"},
     method = RequestMethod.PUT
   )
-  default ResponseEntity<UserResponse> authoritiesAuthorityIdUsersUserIdPut(
+  default ResponseEntity<UserResponse> updateUser(
```
Comment (mgilgar) [inline service/src/generated/java/uk/gov/dft/bluebadge/service/usermanagement/controller/UsersApi.java]: "updateUser" I love that.

File: service/src/main/java/uk/gov/dft/bluebadge/service/usermanagement/UsersApiControllerImpl.java [hunk 3]
```diff
@@ -125,26 +123,16 @@ public UsersApiControllerImpl(
...
-      Error error = new Error();
-      for (ErrorErrors errorItem : e.getErrorsList()) {
-        error.addErrorsItem(errorItem);
-      }
-      userResponse.setError(error);
-      return ResponseEntity.badRequest().body(userResponse);
-    }
+
+    int result = service.createUser(entity);
+    userResponse.setData(userConverter.convertToData(entity, 1, result, 0));
```
Comment (mgilgar) [inline service/src/main/java/uk/gov/dft/bluebadge/service/usermanagement/UsersApiControllerImpl.java]: What is 1 and 0?

File: service/src/main/java/uk/gov/dft/bluebadge/service/usermanagement/service/UserManagementService.java [hunk 4]
```diff
@@ -49,14 +40,25 @@
...
     if (null != businessErrors) {
-      throw new UserEntityValidationException(businessErrors);
+      throw new BadResponseException(businessErrors);
```
Comment (mgilgar) [inline service/src/main/java/uk/gov/dft/bluebadge/service/usermanagement/service/UserManagementService.java]: Is it BadResponseException or BadRequestException?

File: service/src/main/java/uk/gov/dft/bluebadge/service/usermanagement/service/UserManagementService.java [hunk 5]
```diff
@@ -130,22 +112,12 @@ public int deleteUser(int id) {
...
-    return matcher.find();
-  }
-
   /**
    * Update user entity.
    *
    * @param userEntity Entity to update.
    * @return Update count.
-   * @throws BlueBadgeBusinessException if validation fails.
+   * @throws BadResponseException if validation fails.
```
Comment (mgilgar) [inline service/src/main/java/uk/gov/dft/bluebadge/service/usermanagement/service/UserManagementService.java]: Maybe BadRequestException?

File: service/src/main/java/uk/gov/dft/bluebadge/service/usermanagement/UsersApiControllerImpl.java (hunk 3 above)
Comment (pauljroberts) [reply service/src/main/java/uk/gov/dft/bluebadge/service/usermanagement/UsersApiControllerImpl.java]: They will be removed when we refactor the error handling.

File: service/src/main/java/uk/gov/dft/bluebadge/service/usermanagement/service/UserManagementService.java (hunk 5 above)
Comment (pauljroberts) [reply service/src/main/java/uk/gov/dft/bluebadge/service/usermanagement/service/UserManagementService.java]: Applied change.

**Pull request merged**